Gems/EMotionFX/Code/Source/Editor/Plugins/Ragdoll/JointSwingLimitManipulators.cpp

```diff
@@ -0,0 +1,245 @@
+                AZ::Transform::CreateFromQuaternion(m_physicsSetupManipulatorData.m_jointConfiguration->m_parentLocalRotation);
+
+            AZStd::optional<float> swingLimitY =
+                m_physicsSetupManipulatorData.m_jointConfiguration->GetPropertyValue(AZ::Name("SwingLimitY"));
+            if (swingLimitY.has_value())
+            {
+                m_swingYManipulator->SetSpace(parentWorldTransform);
+                m_swingYManipulator->SetLocalPosition(AZ::Vector3(ManipulatorOffsetX, 0.0f, ManipulatorInverseScale * swingLimitY.value()));
+            }
+            AZStd::optional<float> swingLimitZ =
+                m_physicsSetupManipulatorData.m_jointConfiguration->GetPropertyValue(AZ::Name("SwingLimitZ"));
+            if (swingLimitZ.has_value())
+            {
+                m_swingZManipulator->SetSpace(parentWorldTransform);
+                m_swingZManipulator->SetLocalPosition(AZ::Vector3(ManipulatorOffsetX, ManipulatorInverseScale * swingLimitZ.value(), 0.0f));
+            }
+        }
+    }
+
+    void JointSwingLimitManipulators::Teardown()
+    {
+        PhysicsSetupManipulatorRequestBus::Handler::BusDisconnect();
+        AZ::TickBus::Handler::BusDisconnect();
+        m_swingYManipulator->Unregister();
+        m_swingZManipulator->Unregister();
+        m_debugDisplay = nullptr;
+    }
+
+    void JointSwingLimitManipulators::ResetValues()
+    {
+        if (m_physicsSetupManipulatorData.HasJointLimit())
+        {
+            BeginEditing();
+            m_physicsSetupManipulatorData.m_jointConfiguration->SetPropertyValue(AZ::Name("SwingLimitY"), 45.0f);
+            m_physicsSetupManipulatorData.m_jointConfiguration->SetPropertyValue(AZ::Name("SwingLimitZ"), 45.0f);
+            FinishEditing();
+            Refresh();
+        }
+    }
+
+    void JointSwingLimitManipulators::InvalidateEditorValues()
+    {
+        if (m_physicsSetupManipulatorData.m_jointLimitWidget)
+        {
+            m_physicsSetupManipulatorData.m_jointLimitWidget->InvalidateValues();
+        }
+    }
+
+    void JointSwingLimitManipulators::OnUnderlyingPropertiesChanged()
+    {
+        Refresh();
+    }
+
+    void JointSwingLimitManipulators::BeginEditing()
+    {
+        m_jointSwingLimitState.m_swingLimitY =
+            m_physicsSetupManipulatorData.m_jointConfiguration->GetPropertyValue(AZ::Name("SwingLimitY"));
+        m_jointSwingLimitState.m_swingLimitZ =
+            m_physicsSetupManipulatorData.m_jointConfiguration->GetPropertyValue(AZ::Name("SwingLimitZ"));
+
+        CreateCommandAdjustJointLimit(m_commandGroup, m_physicsSetupManipulatorData);
+    }
+
+    void JointSwingLimitManipulators::FinishEditing()
+    {
+        ExecuteCommandAdjustJointLimit(m_commandGroup, m_physicsSetupManipulatorData);
+    }
+
+    void JointSwingLimitManipulators::OnTick([[maybe_unused]] float delta, [[maybe_unused]] AZ::ScriptTimePoint timePoint)
```

Comment:
suggestion: Is it possible to use a DebugDisplay notification bus here instead of OnTick? That is generally preferred but if an appropriate one doesn't exist we can leave this for this PR - might be something we should make a task for though (so you can connect on ViewportId and get the callback at the right time)

Reply:
yeah, don't think it works the same way as the editor currently, but we can look at adding that later